llama_index/storage/index_store/pickled_keyval_index_store.py

```diff
@@ -0,0 +1,75 @@
+from typing import List, Optional
+from llama_index.data_structs.data_structs import IndexStruct
+from llama_index.storage.index_store.types import BaseIndexStore
+from llama_index.constants import DATA_KEY, TYPE_KEY
+from llama_index.storage.kvstore.simple_pickled_kvstore import SimplePickledKVStore
+
+DEFAULT_NAMESPACE = "index_store"
+
+
+class PickledKVIndexStore(SimplePickledKVStore):
+    """Key-Value Index store that uses pickle as a storing mechanism.
+
+    Args:
+        kvstore (BaseKVStore): key-value store
+        namespace (str): namespace for the index store
+
+    """
+
+    def __init__(self, kvstore: SimplePickledKVStore, namespace: Optional[str] = None) -> None:
+        """Init a PickledKVIndexStore."""
+        self._kvstore = kvstore
+        self._namespace = namespace or DEFAULT_NAMESPACE
+        self._collection = f"{self._namespace}/data"
+
+    def add_index_struct(self, index_struct: IndexStruct) -> None:
```

Comment:
High level comment, it seems like a lot of the code here is duplicated. I wonder if we can just extend the original simple index store to just modify the underlying kvstore?

Reply:
I agree. I'll rework this when I have time.